test/e2e/network/endpointslice.go

```diff
@@ -299,6 +303,146 @@ var _ = SIGDescribe("EndpointSlice", func() {
 		deleteEndpointSlices(cs, f.Namespace.Name, svc2)
 		expectEndpointsAndSlices(cs, f.Namespace.Name, svc2, []*v1.Pod{pod1, pod2}, 2, 2, true)
 	})
+
+	// Ensure that garbage collection works well when a Service is deleted and
+	// its EndpointSlices are being rapidly updated.
+	// For more context: https://github.com/kubernetes/kubernetes/issues/91287
+	ginkgo.It("should not get in the way of Service owner garbage collection", func() {
+		sharedLabels := map[string]string{"test-garbage-collection": "for-service"}
+
+		deployment, err := cs.AppsV1().Deployments(f.Namespace.Name).Create(context.TODO(), &appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   "test-garbage-collection",
+				Labels: sharedLabels,
+			},
+			Spec: appsv1.DeploymentSpec{
+				Selector: &metav1.LabelSelector{MatchLabels: sharedLabels},
+				Replicas: utilpointer.Int32Ptr(1),
+				Template: v1.PodTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: sharedLabels,
+					},
+					Spec: v1.PodSpec{
+						Containers: []v1.Container{
+							{
+								Name:  "container1",
+								Image: imageutils.GetE2EImage(imageutils.Nginx),
```

Comment:
nginx was not ipv6 friendly (it does not listen in IPv6 by default)
, but if we don't have to poll the port, should not be better using busybox or a lighter image?

Reply:
yeah, just a pause image seems workable if we're not actually exercising the network

Reply:
Updated, thanks!